site-src/geps/gep-1364.md

```diff
@@ -0,0 +1,328 @@
+path from "outside"/lacking cluster context to "inside"/enriched with cluster context,
+we also need a way to describe when that data path is configured and working.
+
+Again, we already have a word we use for this - "Ready", so this GEP proposes keeping
+that and tightening up the definition.
+
+### Condition polarity
+
+In terms of the polarity of conditions, we have three options, of which only two are
+really viable:
+* All conditions must be negative polarity
+* All conditions must be positive polarity
+* Some conditions can be positive polarity, but most should be negative.
+
+The fact that the user experience of `Ready` or conditions like `Accepted` being `true`
+in the healthy case is much better rules out the first option, so we are left to
+decide between enforcing that all conditions are positive, or that we have a mix.
+
+Having an arbitrary mix will make doing machine-based extraction of information
+much harder, so here I'm going to talk about the distinction between having all
+conditions positive or some, summary conditions positive, and the rest negative.
+
+#### All Conditions Positive
+
+In this case, all Condition types are written in such a way that they're positive
+polarity, and are `true` in the healthy case.
+
+As already discussed, `Ready`, and `Accepted` are current examples, but another
+one that's a little more important here is `ResolvedRefs` which is set to `true`
+when all references to other resources have been successfully resolved. This is
+not a _blocking_ Condition that affects the `Ready` condition, since having _some_
+references valid is enough to produce some configuration in the underlying data
+plane.
+
+So, All Conditions Positive pros:
+* We're close already. Most conditions in the API are currently positive polarity.
+* Easier to understand - there are no double negatives. "Good: true" is less
+cognitive overhead than "NotGood: false".
+
+Cons:
+* Reduces flexibility - it can surprisingly difficult to avoid double negatives for
+conditions that describe error states, as in general programmers are more used
+to reporting "something went wrong" than they are "everything's okay".
+
+Not sure if pro or con:
+* Leans the design towards favoring conditions always being present, since you
+can't be sure if everything is good unless you see `AllGood: true`. The absence
+of a positive-polarity condition implies that the condition could be false. This
+puts this option more in line with the API guidelines on this point.
+
+#### Some Conditions Positive
+
+In this case, only a limited set of summary conditions are positive, and the rest
+are negative.
+
+Pros:
+* Error states can be described with `Error: true` instead of `NoError: false`.
+* Negative polarity error conditions are more friendly to not being present (since
+absence of `Error: true` implies everything's okay).
+
+Cons:
+* Double negatives in the happy case: `NoError: false`
```

Comment:
This seems like a con of "All Conditions Positive" but not sure it applies when we allow both positive and negative conditions?

Reply:
Ah yes, you're right.